src/Clash/Cores/ClashFloPoCo/GenTemDSL.hs

```diff
@@ -0,0 +1,1306 @@
+import Data.Text.Prettyprint.Doc.Extra (Doc)
+import GHC.Stack (HasCallStack)
+import Language.Haskell.TH.Syntax
+import Text.Show.Pretty (ppShow)
+import Prelude
+
+-- | Function to convert pipeline depth from InfoEntity to Nat number
+getPipeDep :: InfoEntity -> Q Type
+getPipeDep infoen = do
+  return (LitT (NumTyLit (fromIntegral (convertMaybeToInt (pipedep infoen)))))
+
+-- | Function to generate the InfoEntity coresponding to the arithmetic entity generated by FloPoCo
+--
+--  __Example:__
+--
+--  > floPoCoPath = "/home/user/flopoco/build/bin/flopoco"
+--  > args = ["frequency=100", "target=Zynq7000", "IEEEFPAdd", "wE=8", "wF=23","name=plusFloat", "outputFile=flopocoAdd.vhdl","registerLargeTables=1"]
+--  > filePath = "flopocoAdd.vhdl"
+--  > infoEnplus = $(genFloPoCoInfoEntity floPoCoPath args filePath)
+genFloPoCoInfoEntity ::
+  -- | path where FloPoCo is built or installed
+  String ->
+  -- | input to feed FloPoCo
+  [String] ->
+  -- | Path to the VHDL file generated by FloPoCo
+  String ->
+  Q Exp
+genFloPoCoInfoEntity floPoCoPathName argsName fileName = do
+  output <- liftIO $ FPCC.callFloPoCoWithInput floPoCoPathName argsName ""
+  if "Error" `elem` lines output
+    then fail "Error in FloPoCo output"
+    else do
+      result <- processVHDLFile getLastInfoEntity fileName
+      liftIO $ print result
+      case result of
+        Just infoentity -> [|$(liftInfoEntity infoentity)|] -- Lift InfoEntity as expression
+        Nothing -> fail "Failed to generate InfoEntity"
+
+-- | Helper function to lift InfoEntity to Q Exp (used internally by genFloPoCoInfoEntity)
+liftInfoEntity :: InfoEntity -> Q Exp
+liftInfoEntity (InfoEntity n f p ins outs) =
+  [|InfoEntity $(lift n) $(lift f) $(lift p) $(lift ins) $(lift outs)|]
+
+--
+--
+
+-- | Function to generate HDL Primitive for clash
+--
+--  Please see the source file if cannot see the content of the example
+--
+--  ==== __Example:__
+--
+--  > {-# ANN plusFloat (flopocoPrim 'plusFloat 'plusFloatBBF) #-}
+--  is equavilent to
+--
+--  > {-# ANN plusFloat (
+--  >    let
+--  >      primName = show 'plusFloat
+--  >      tfName = show 'plusFloatBBF
+--  >   in
+--  >      InlineYamlPrimitive [minBound..] [__i|
+--  >        BlackBoxHaskell:
+--  >          name: #{primName}
+--  >          templateFunction: #{tfName}
+--  >          workInfo: Always
+--  >      |]) #-}
+flopocoPrim ::
+  -- | Name of the Clash function corresponding to entity(VHDL) or module(Verilog, SystemVerilog)
+  Name ->
+  -- | Name of BlackBox function of the Clash function
+  Name ->
+  Primitive
+flopocoPrim fprimName ftfName =
+  let primName = fprimName
+      tfName = ftfName
+   in InlineYamlPrimitive
+        [minBound ..]
+        [__i|
+        BlackBoxHaskell:
+          name: #{primName}
+          templateFunction: #{tfName}
+          workInfo: Always
+        |]
+
+-- | Function to convert Maybe Int to Int
+convertMaybeToInt :: Maybe Int -> Int
+convertMaybeToInt = fromMaybe 0

```

Comment:
I would just inline this definition, because:

1. It only has one use site
2. The name `convertMaybeToInt` doesn't really inform the user that happens when the argument is `Nothing`; the choice for `0` is arbitrary.